Feedback after ml changes

frimpongopoku opened the pull request:
## Related ticket : https://github.com/massenergize/frontend-admin/issues/1100 ,  https://github.com/orgs/massenergize/projects/7/views/2?filterQuery=media&pane=issue&itemId=40748319 ,  https://github.com/massenergize/frontend-admin/issues/1102
## Works with backend PR here:

frimpongopoku commented:
@BradHN1  Question 

In this ticket: https://github.com/massenergize/frontend-admin/issues/1102 
I dont know what you mean by "Uploaded by community [by default, if superadmin, select all communities, if community admin select my communities.]" 

Are you saying I should show the full list of available communities to community admins? Then put special focus on their communities? 

Cos currently search by community for community admins only shows the list of communities they manage in the list, so there is no point in putting any special focus.... does it mean we want to show all the communities to them? 


This is what I get as a community admin, I get to choose only from the list of communities I manage. Only super admins get to choose from the full list. Should I change that? Is that what you are asking for? 
![Screenshot 2023-10-11 at 10 23 17](https://github.com/massenergize/frontend-admin/assets/26961591/6c835c13-510b-4712-bfcc-311fb040bd52)

BradHN1 commented:
I think now it is just called “Community” which is default, but it really means the images which were uploaded by particular communities.  As opposed to owned by or created by those communities.  
It is the default choice, and if you are a superadmin users it should have all communities selected; if you are a cadmin you should have your communities selected.

I assume that if I’m a cadmin of say two communities I may be able to see images uploaded by some other communities so I can select them and use those images, is that right?  If a cadmin says make an image available to all communities, then it should be findable by all community admins.

The tricky thing might be if I create an action with an image (maybe it’s a massenergise template) and other communities use it, who uploaded it?  I guess that the MassEnergize Template community uploaded it.

Happy to talk further.  Perfection isn’t the goal here.


> On Oct 11, 2023, at 6:25 AM, Frimpong Opoku Agyemang ***@***.***> wrote:
> 
> 
> @BradHN1 <https://github.com/BradHN1> Question
> 
> In this ticket: #1102 <https://github.com/massenergize/frontend-admin/issues/1102>
> I dont know what you mean by "Uploaded by community [by default, if superadmin, select all communities, if community admin select my communities.]"
> 
> Are you saying I should show the full list of available communities to community admins? Then put special focus on their communities?
> 
> Cos currently search by community for community admins only shows the list of communities they manage in the list, so there is no point in putting any special focus.... does it mean we want to show all the communities to them?
> 
> This is what I get as a community admin, I get to choose only from the list of communities I manage. Only super admins get to choose from the full list. Should I change that? Is that what you are asking for?
>  <https://user-images.githubusercontent.com/26961591/274231805-6c835c13-510b-4712-bfcc-311fb040bd52.png>
> —
> Reply to this email directly, view it on GitHub <https://github.com/massenergize/frontend-admin/pull/1104#issuecomment-1757346186>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADBVDZANSHZQE4EMAOAVSIDX6ZXYBANCNFSM6AAAAAA53XIYMU>.
> You are receiving this because you were mentioned.
>

frimpongopoku commented:
> I think now it is just called “Community” which is default, but it really means the images which were uploaded by particular communities. As opposed to owned by or created by those communities. It is the default choice, and if you are a superadmin users it should have all communities selected; if you are a cadmin you should have your communities selected. I assume that if I’m a cadmin of say two communities I may be able to see images uploaded by some other communities so I can select them and use those images, is that right? If a cadmin says make an image available to all communities, then it should be findable by all community admins. The tricky thing might be if I create an action with an image (maybe it’s a massenergise template) and other communities use it, who uploaded it? I guess that the MassEnergize Template community uploaded it. Happy to talk further. Perfection isn’t the goal here.
> […](#)
> On Oct 11, 2023, at 6:25 AM, Frimpong Opoku Agyemang ***@***.***> wrote: @BradHN1 <https://github.com/BradHN1> Question In this ticket: #1102 <#1102> I dont know what you mean by "Uploaded by community [by default, if superadmin, select all communities, if community admin select my communities.]" Are you saying I should show the full list of available communities to community admins? Then put special focus on their communities? Cos currently search by community for community admins only shows the list of communities they manage in the list, so there is no point in putting any special focus.... does it mean we want to show all the communities to them? This is what I get as a community admin, I get to choose only from the list of communities I manage. Only super admins get to choose from the full list. Should I change that? Is that what you are asking for? <https://user-images.githubusercontent.com/26961591/274231805-6c835c13-510b-4712-bfcc-311fb040bd52.png> — Reply to this email directly, view it on GitHub <[#1104 (comment)](https://github.com/massenergize/frontend-admin/pull/1104#issuecomment-1757346186)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ADBVDZANSHZQE4EMAOAVSIDX6ZXYBANCNFSM6AAAAAA53XIYMU>. You are receiving this because you were mentioned.



@BradHN1  one of the requirements in the doc was to allow admins to mark an upload as "to all communities" (public). 
That is why I introduced the "publicity" field on the backend, and also added the "public" field as a filter option. With this as a feature, any image that is outside of a community admins's communities can be found there, as long as whoever uploaded it marked it as public. Isnt that a cleaner approach? Since if we allow a community admin to see all of the available communities and allow them to filter uploads based other communities, they might end up using media they are not allowed to use.  🤔 



And about the template example you gave earlier, I dont think it will be that tricky. When you create a template as a super admin or community admin you are still a user, and you need to specify the allowed communities or mark any attached media as public. There is that, and there is also the fact that the upload you make is also always related to you as a signed in user. 
Knowing the way things are structured, this is what I think will happen in your example: 
After you have created your template, any copies made by any communities will have the attached media in place regardless of whether their communities are allowed to use it or not.  As long as admins dont tinker with the image, they will still have it in place. However, if a cadmin changes the image after they have made a copy, and it happens that their community is not listed in the allowed list, they will definitely not find that particular media in the library anywhere to ever use again -- and this I think is consistent  and normal behaviour. 

This is why it will probably be best to mark a media as public when creating a template 🤣.

BradHN1 commented:
In testing locally, this blows up when I try to set the pictures on a home page from the gallery.
Lots of warnings, but here is the first error in console:
<img width="708" alt="Screenshot 2023-10-16 at 6 03 15 PM" src="https://github.com/massenergize/frontend-admin/assets/12800484/9d98a63c-9add-4269-94be-2d157fd8f9b2">

frimpongopoku commented:
> In testing locally, this blows up when I try to set the pictures on a home page from the gallery. Lots of warnings, but here is the first error in console: <img alt="Screenshot 2023-10-16 at 6 03 15 PM" width="708" src="https://user-images.githubusercontent.com/12800484/275660160-9d98a63c-9add-4269-94be-2d157fd8f9b2.png">

@BradHN1 you can retry. It should be fixed now 🤔

BradHN1 reviewed (approved):
LGTM
